runelite-client/src/main/java/net/runelite/client/game/chatbox/ChatboxItemSearch.java

```diff
@@ -0,0 +1,193 @@
+	private static final int FONT_SIZE = 16;
+
+	private final ChatboxPanelManager chatboxPanelManager;
+	private final ItemManager itemManager;
+	private final Client client;
+
+	private HashMap<String, ItemComposition> results = new LinkedHashMap<>();
+	private String tooltipText;
+
+	@Inject
+	public ChatboxItemSearch(ChatboxPanelManager chatboxPanelManager, ClientThread clientThread,
+		ItemManager itemManager, Client client)
+	{
+		super(chatboxPanelManager, clientThread);
+		this.chatboxPanelManager = chatboxPanelManager;
+		this.itemManager = itemManager;
+		this.client = client;
+
+		lines(1);
+		prompt("Item Search");
+		onChanged(searchString ->
+		{
+			clientThread.invokeLater(() ->
+			{
+				filterResults();
+				update();
+			});
+		});
+	}
+
+	@Override
+	protected void update()
+	{
+		Widget container = chatboxPanelManager.getContainerWidget();
+		container.deleteAllChildren();
+
+		Widget promptWidget = container.createChild(-1, WidgetType.TEXT);
+		promptWidget.setText(getPrompt());
+		promptWidget.setTextColor(0x800000);
+		promptWidget.setFontId(getFontID());
+		promptWidget.setOriginalX(0);
+		promptWidget.setOriginalY(5);
+		promptWidget.setXPositionMode(WidgetPositionMode.ABSOLUTE_CENTER);
+		promptWidget.setYPositionMode(WidgetPositionMode.ABSOLUTE_TOP);
+		promptWidget.setOriginalHeight(FONT_SIZE);
+		promptWidget.setXTextAlignment(WidgetTextAlignment.CENTER);
+		promptWidget.setYTextAlignment(WidgetTextAlignment.CENTER);
+		promptWidget.setWidthMode(WidgetSizeMode.MINUS);
+		promptWidget.revalidate();
+
+		buildEdit(0, 5 + FONT_SIZE, container.getWidth(), FONT_SIZE);
+
+		Widget separator = container.createChild(-1, WidgetType.LINE);
+		separator.setOriginalX(0);
+		separator.setOriginalY(8 + (FONT_SIZE * 2));
+		separator.setXPositionMode(WidgetPositionMode.ABSOLUTE_CENTER);
+		separator.setYPositionMode(WidgetPositionMode.ABSOLUTE_TOP);
+		separator.setOriginalHeight(0);
+		separator.setOriginalWidth(16);
+		separator.setWidthMode(WidgetSizeMode.MINUS);
+		separator.setTextColor(0x666666);
+		separator.revalidate();
+
+		int x = PADDING;
+		int y = PADDING * 3;
+
+		for (ItemComposition itemComposition : results.values())
+		{
+			Widget item = container.createChild(-1, WidgetType.GRAPHIC);
+			item.setXPositionMode(WidgetPositionMode.ABSOLUTE_LEFT);
+			item.setYPositionMode(WidgetPositionMode.ABSOLUTE_TOP);
+			item.setOriginalX(x);
+			item.setOriginalY(y + FONT_SIZE * 2);
+			item.setOriginalHeight(ICON_HEIGHT);
+			item.setOriginalWidth(ICON_WIDTH);
+			item.setName("<col=ff9040>" + itemComposition.getName());
+			item.setItemId(itemComposition.getId());
+			item.setItemQuantity(10000);
+			item.setItemQuantityMode(ItemQuantityMode.NEVER);
+			item.setBorderType(1);
+			item.setAction(0, tooltipText);
+			item.setHasListener(true);
+			item.setOnOpListener((JavaScriptCallback) ev ->
+			{
+				if (getOnDone() != null)
+				{
+					getOnDone().accept(itemComposition.getId() + "");
```

Comment:
override the existing onDone and mark it deprecated, then define a new method+field, maybe onItemSelected so you don't have to make it a string then parse that again